app/controllers/challenge_signups_controller.rb

```diff
@@ -322,4 +323,54 @@ def prompt_meme_to_csv
 
     csv_array
   end
+
+  private
+
+  def challenge_signup_params
+    params.require(:challenge_signup).permit(
+      :pseud_id,
+      requests_attributes: nested_prompt_params,
+      offers_attributes: nested_prompt_params
+    )
+  end
+
+  def nested_prompt_params
+    [
+      :id,
+      :collection_id,
+      :title,
+      :url,
+      :any_fandom,
+      :any_character,
+      :any_relationship,
+      :any_freeform,
+      :any_category,
+      :any_rating,
+      :any_warning,
+      :anonymous,
+      :description,
+      :_destroy,
+      tag_set_attributes: [
+        :id,
```

Comment:
Oh, thank you, that's all good to know! And also kind of frustrating that it behaves so unexpectedly, blargh.

Because it's not really clear what the best course of action is here, I think we can just go with the quickest -- I'm going to merge this now, and we'll just keep watch for any weirdness around the prompts controller as we test.